objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,40 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)getAnyTypeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix], @"Invalid any type url.");
+  return [self.typeURL substringFromIndex: [GPBTypeGoogleApisComPrefix length]];
+}
+
+- (void)unpackTo:(GPBMessage*)message {
+  NSAssert([self is:[message class]],
+           [NSString stringWithFormat:@"%@/%@/%@/%@/%@",
+                    @"Given messsage's type name (",
+                    message.descriptor.name,
+                    ") doesn't match with any's (",
+                    self.getAnyTypeName,
+                    ")"]);
+  [message mergeFromData:self.value extensionRegistry:NULL];
+}
+
+- (void) packFrom:(GPBMessage*)message {
```

Comment:
What are you doing with this object afterwards? Realize that to store it somewhere you will need to copy it. I doubt you are going to want to reuse, and would probably instead be creating multiple.